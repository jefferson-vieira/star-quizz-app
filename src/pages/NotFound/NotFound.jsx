import React from 'react';
import { Link } from 'react-router-dom';

import Header from '../../components/Header';

const NotFound = () => (
  <section id="not-found" class="not-found container-fluid">
    <Header />
    <section className="container my-5">
      <header class="not-found__header border-warning border-bottom pb-2 my-3">
        <h1>Ops...</h1>
      </header>
      <div className="not-found__message my-5">
        <h2>Página não encontrada.</h2>
        <h3>Clique no botão abaixo para ir até a página inicial.</h3>
      </div>
      <div class="row justify-content-center">
        <Link
          to="/home"
          className="btn btn-outline-warning btn-lg"
          title="Home"
        >
          Home
        </Link>
      </div>
    </section>
  </section>
);

export default NotFound;
