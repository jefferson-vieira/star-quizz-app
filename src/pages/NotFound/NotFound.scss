@import 'assets/scss/variables';

.not-found {
  &__message {
    color: $white;
  }
}
