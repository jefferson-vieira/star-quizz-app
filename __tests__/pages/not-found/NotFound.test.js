import React from 'react';
import { shallow } from 'enzyme';

import NotFound from '../../../src/pages/NotFound/NotFound';

describe('Testing NotFound Component...', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<NotFound />);
    expect(wrapper).toMatchSnapshot();
  });
});
