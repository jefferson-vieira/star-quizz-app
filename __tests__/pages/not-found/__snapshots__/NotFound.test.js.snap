// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NotFound Component... should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PageHeader />,
        <section
          className="container my-5"
        >
          <Header
            title="Ops..."
          />
          <div
            className="not-found__message my-5"
          >
            <h2>
              Página não encontrada.
            </h2>
            <h3>
              Clique no botão abaixo para ir até a página inicial.
            </h3>
          </div>
          <HomeButton />
        </section>,
      ],
      "className": "not-found container-fluid",
      "id": "not-found",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Header
              title="Ops..."
            />,
            <div
              className="not-found__message my-5"
            >
              <h2>
                Página não encontrada.
              </h2>
              <h3>
                Clique no botão abaixo para ir até a página inicial.
              </h3>
            </div>,
            <HomeButton />,
          ],
          "className": "container my-5",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "title": "Ops...",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Página não encontrada.
                </h2>,
                <h3>
                  Clique no botão abaixo para ir até a página inicial.
                </h3>,
              ],
              "className": "not-found__message my-5",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Página não encontrada.",
                },
                "ref": null,
                "rendered": "Página não encontrada.",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Clique no botão abaixo para ir até a página inicial.",
                },
                "ref": null,
                "rendered": "Clique no botão abaixo para ir até a página inicial.",
                "type": "h3",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PageHeader />,
          <section
            className="container my-5"
          >
            <Header
              title="Ops..."
            />
            <div
              className="not-found__message my-5"
            >
              <h2>
                Página não encontrada.
              </h2>
              <h3>
                Clique no botão abaixo para ir até a página inicial.
              </h3>
            </div>
            <HomeButton />
          </section>,
        ],
        "className": "not-found container-fluid",
        "id": "not-found",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Header
                title="Ops..."
              />,
              <div
                className="not-found__message my-5"
              >
                <h2>
                  Página não encontrada.
                </h2>
                <h3>
                  Clique no botão abaixo para ir até a página inicial.
                </h3>
              </div>,
              <HomeButton />,
            ],
            "className": "container my-5",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "title": "Ops...",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2>
                    Página não encontrada.
                  </h2>,
                  <h3>
                    Clique no botão abaixo para ir até a página inicial.
                  </h3>,
                ],
                "className": "not-found__message my-5",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Página não encontrada.",
                  },
                  "ref": null,
                  "rendered": "Página não encontrada.",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Clique no botão abaixo para ir até a página inicial.",
                  },
                  "ref": null,
                  "rendered": "Clique no botão abaixo para ir até a página inicial.",
                  "type": "h3",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
